widget/embedded/src/types/event.ts

```diff
@@ -0,0 +1,76 @@
+import type { SelectedQuote } from './quote';
+import type { Wallet } from './wallets';
+import type {
+  RouteEventData,
+  StepEventData,
+} from '@rango-dev/queue-manager-rango-preset';
+
+import { MainEvents as QueueManagerEvents } from '@rango-dev/queue-manager-rango-preset';
+
+type Account = Wallet;
+
+export enum QuoteEventTypes {
+  QUOTE_INPUT_UPDATE = 'quoteInputUpdate',
+  QUOTE_UPDATE = 'quoteUpdate',
+}
+
+export enum WalletEventTypes {
+  CONNECT = 'connect',
+  DISCONNECT = 'disconnect',
+}
+
+export type QuoteInputUpdateEventPayload = {
+  fromBlockchain?: string;
+  toBlockchain?: string;
+  fromToken?: { symbol: string; name: string | null; address: string | null };
+  toToken?: { symbol: string; name: string | null; address: string | null };
+  requestAmount?: string;
+};
+
+export type QuoteUpdateEventPayload = Pick<
+  SelectedQuote,
+  'requestAmount' | 'swaps' | 'outputAmount' | 'resultType' | 'tags'
+> | null;
+
+export type ConnectWalletEventPayload = {
+  walletType: string;
+  accounts: Account[];
+};
+
+export type DisconnectWalletEventPayload = {
+  walletType: string;
+};
+
+export type QuoteEventData =
+  | {
+      type: QuoteEventTypes.QUOTE_INPUT_UPDATE;
+      payload: QuoteInputUpdateEventPayload;
+    }
+  | {
+      type: QuoteEventTypes.QUOTE_UPDATE;
+      payload: QuoteUpdateEventPayload;
+    };
+
+export type WalletEventData =
+  | {
+      type: WalletEventTypes.CONNECT;
+      payload: ConnectWalletEventPayload;
+    }
+  | {
+      type: WalletEventTypes.DISCONNECT;
+      payload: DisconnectWalletEventPayload;
+    };
+
+export enum MainEvents {
+  RouteEvent = QueueManagerEvents.RouteEvent,
+  StepEvent = QueueManagerEvents.StepEvent,
+  QuoteEvent = 'quoteEvent',
+  WalletEvent = 'walletEvent',
+}
+
+export type Events = {
+  [MainEvents.RouteEvent]: RouteEventData;
+  [MainEvents.StepEvent]: StepEventData;
+  [MainEvents.QuoteEvent]: QuoteEventData;
+  [MainEvents.WalletEvent]: WalletEventData;
+};
```

Comment:
Please Make sure all related types for using widget events including `StepEventType`, `RouteEventType`, `WalletEventTypes`, `QuoteEventTypes`, `WidgetEvent`, `RouteEventData`, `StepEventData`, `QuoteEventData`, `WalletEventData`, ... are exported from the widget embedded package root.
I'm not sure if I've mentioned all required types but these are some samples.

Reply:
Done